pkg/kubectl/cmd/apply_set_last_applied.go

```diff
@@ -0,0 +1,169 @@
+	Factory                      cmdutil.Factory
+	Out                          io.Writer
+	ErrOut                       io.Writer
+}
+
+var (
+	applySetLastAppliedLong = templates.LongDesc(`
+		Set the latest last-applied-configuration annotations by file.`)
+
+	applySetLastAppliedExample = templates.Examples(`
+		# Set the current file's content to the latest-applied-configuration of this resource.
+		kubectl apply set-last-applied -f deploy.yaml`)
+)
+
+func NewCmdApplySetLastApplied(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	options := &SetLastAppliedOptions{Out: out, ErrOut: err}
+	cmd := &cobra.Command{
+		Use:     "set-last-applied -f FILENAME",
+		Short:   "Set latest last-applied-configuration annotations of by file",
+		Long:    applySetLastAppliedLong,
+		Example: applySetLastAppliedExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.RunApplySetLastApplied(f, args, cmd, out, err, options))
+		},
+	}
+
+	cmdutil.AddDryRunFlag(cmd)
+	cmdutil.AddRecordFlag(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "that contains the last-applied-configuration annotations"
+	kubectl.AddJsonFilenameFlag(cmd, &options.FilenameOptions.Filenames, "Filename, directory, or URL to files "+usage)
+
+	return cmd
+}
+
+func (o *SetLastAppliedOptions) RunApplySetLastApplied(f cmdutil.Factory, args []string, cmd *cobra.Command, out, errOut io.Writer, options *SetLastAppliedOptions) error {
+	mapper, typer, err := f.UnstructuredObject()
+	if err != nil {
+		return err
+	}
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		Latest().
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	dryRun := cmdutil.GetFlagBool(cmd, "dry-run")
+	output := cmdutil.GetFlagString(cmd, "output")
+	shortOutput := output == "name"
+	codec := f.JSONEncoder()
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		objMap := map[string]map[string]map[string]string{}
+		metadataMap := map[string]map[string]string{}
+		annotationsMap := map[string]string{}
+
+		localFile, err := runtime.Encode(codec, info.VersionedObject)
+		if err != nil {
+			return err
+		}
+		annotationsMap[annotations.LastAppliedConfigAnnotation] = string(localFile)
+		metadataMap["annotations"] = annotationsMap
+		objMap["metadata"] = metadataMap
+		jsonString, err := json.Marshal(objMap)
+		if err != nil {
+			return err
+		}
+
+		if err := info.Get(); err != nil {
+			if errors.IsNotFound(err) {
+				return err
+			} else {
+				return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+			}
+		}
+
+		if !dryRun {
```

Comment:
`dry-run` means just print the object that would be sent~yep that would make sense print out the patch when dry-run